src/view/RepeatedFragment.js

```diff
@@ -210,6 +210,10 @@ export default class RepeatedFragment {
 			if ( newIndex === -1 ) return;
 
 			const fragment = this.iterations[ oldIndex ];
+
+			// guard against null fragment - #2005 (feels like treating the symptom, not the cause?)
+			if ( fragment == null ) return;
```

Comment:
@evs-chris Indeed, the wrapper and model gets out of sync. I spent some debugging and found the following sequence:

We start by doing childrenArray.push(...)
1. we call a function, ```processWrapper```, which retrieves it's associated ractive model through the hack: ```var __model = wrapper.__model;```
2. we then shuffle the model, shuffle the fragments and eventualy call ```RepeatedFragment.updatePostShuffle``` to rebind the model to the iterated fragments
3. we render the new fragment

Next we call parentArray.splice(...)
1.  again we call processWrapper, which retrieves it's associated ractive model through the hack: ```var __model = wrapper.__model;```
2. we then shuffle the model and shuffle the fragments. Because this is a splice we move fragments to different models.
3. we don't seem to update the ```wrapper.__model``` to it's new model, so ```wrapper.__model``` points to it's old model and old fragment

Next we call childrenArray.push(..) again:
1. same story, ```processWrapper``` retrieves it's associated ractive model through the hack: ```var __model = wrapper.__model;```. This model is still the old model pointing to the old fragment
2. we then do shuffling and eventually ```RepeatedFragment.updatePostShuffle``` to rebind the model to the iterated fragments. However because we have the wrong model, we also have the wrong RepeatedFragment instance (this fragment has no iterations) and retrieves the null fragment.

Hope that makes sense.

So the fix requires us to update the ```wrapper.__model``` once shuffling occurs. But I'm not sure where in the code that should occur?

The wrapper.__model is hacky anyway, so probably needs a bit of a overhaul to be more robust?